### ISSUE TITLE
Enable KDE integration in LibreOffice

### ISSUE THREAD

**Issue by vandenoever** [NixOS/nixpkgs#73387]
**Project description**

LibreOffice has new KDE integration. This can be enabled with `--enable-qt5 --enable-kde5`.

Enabling these flags lets LibreOffice detect the appropriate integration. On Plasma, it will use Qt widgets. On GNOME it will use GTK widgets.

Here is the latest talk describing this improvement.
https://conf.linuxappsummit.org/en/LAS2019/public/events/60

**Metadata**

* homepage URL: https://www.libreoffice.org/
* source URL: https://donate.libreoffice.org/home/dl/src/6.3.3/all/libreoffice-6.3.3.2.tar.xz?idx=1
* license: MPLv2.0 (secondary license GPL, LGPLv3+ or Apache License 2.0)[9]
* platforms: all

**Comment by vandenoever** [comment]
@7c6f434c Pinging the LibreOffice maintainer on this issue.

**Comment by 7c6f434c** [comment]
I am really unlikely to test it (I haven't used KDE nor GNOME for ages), but willing to merge a PR if someone does test this.

**Comment by vandenoever** [comment]
I'm willing to test it, but unsure how to patch it. Should we keep kde4 integration?

**Comment by 7c6f434c** [comment]
Is kde4 still in Nixpkgs? If no, then I think it is OK to replace the integration.

I hope that passing some amount of Qt5 libraries as `buildInputs` and some Qt5 build tools as `nativeBuildInputs` and replacing the integration configure flags should be enough. It is OK to do this only in `default.nix`, `still.nix` gets overwritten by default version when Fresh/Still minor versions get incremented upstream (and we package this update)

**Comment by xbubli** [comment]
IDK what versions of Libreoffice you people package, but kde4 has been deprecated from Libreoffice 6.3 on. In Libreoffice 6.2 both kde4 and kde5 can be built, but are mutually exclusive at runtime so you can't have both. 
But kde5 in 6.2 is in pretty good shape and we keep backporting critical bugfixes (although we haven't had one in months :-)) For example we backported all of the threading bugfixes, so should you people deploy any extensions, they no longer deadlock

**Comment by 7c6f434c** [comment]
On master branch Still is 6.2, Fresh is 6.3. Thanks for the overview.

**Comment by vandenoever** [comment]
I've done some experimenting. So far I cannot get qmake to run properly:
```
Skipping configure process.QMAKEPATH=/nix/store/kv6mipfp13cl4m39c33a7kn0b3yzjs1v-qtbase-5.12.5-dev
qmake PREFIX=/nix/store/pfli681v3iskil5whl0r17q8zifrza46-libreoffice-6.3.3.2 NIX_OUTPUT_OUT=/nix/store/pfli681v3iskil5whl0r17q8zifrza46-libreoffice-6.3.3.2 NIX_OUTPUT_DEV=/nix/store/kxsnvnhqg644by2kyydjvhmqgzv49fyf-libreoffice-6.3.3.2-dev NIX_OUTPUT_BIN=/nix/store/pfli681v3iskil5whl0r17q8zifrza46-libreoffice-6.3.3.2 NIX_OUTPUT_DOC=/nix/store/kxsnvnhqg644by2kyydjvhmqgzv49fyf-libreoffice-6.3.3.2-dev/share/doc/qt-5.12.5 NIX_OUTPUT_QML=/nix/store/pfli681v3iskil5whl0r17q8zifrza46-libreoffice-6.3.3.2/lib/qt-5.12.5/qml NIX_OUTPUT_PLUGIN=/nix/store/pfli681v3iskil5whl0r17q8zifrza46-libreoffice-6.3.3.2/lib/qt-5.12.5/plugins
Usage: qmake [mode] [options] [files]
```

It would be interesting to compare the build output with that of a passing build. Is there a log of a build for some other distro available somewhere?

**Comment by kennyballou** [comment]
Noticed this is a duplicate of #57846 and probably should be closed now or when #68414 lands.

**Comment by vandenoever** [comment]
You are right, it's a duplicate.
